File: Leetcode/1022_Sum_of_Root_To_Leaf_Binary_Numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> v;
    int sumRootToLeaf(TreeNode* root) {
        rootToLeaf(root, "");
        int sum = 0;
        for(unsigned int i = 0; i < v.size(); i++)
        {
            sum += v[i];
        }
        
        return sum;
    }
    
    void rootToLeaf(TreeNode* root, string s)
    {
        if(root == nullptr)
        {
            return;
        }
        if(root->left == nullptr && root->right == nullptr)
        {
            v.push_back(convertIntoDecimal(s + to_string(root->val)));
            return;
        }
        
        rootToLeaf(root->left, s + to_string(root->val));
        rootToLeaf(root->right, s + to_string(root->val));
    }
    
    int convertIntoDecimal(string s)
    {
        int decimal = 0;
        for(int i = 0; i < s.size(); i++)
        {
            decimal = decimal * 2 + s[i] - '0';
        }
        
        return decimal;
    }
};

File: Leetcode/1022_Sum_of_Root_To_Leaf_Binary_Numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumRootToLeaf = function (root) {
    const binaryNumbers = [];
    const sum = [];
    traversal(root, binaryNumbers, sum);

    return sum.reduce((acc, cur) => acc + cur, 0);
};

var traversal = function (root, result, sum) {
    if (root) {
        result.push(root.val);
        traversal(root.left, result, sum);
        if (root.left === null && root.right === null) {
            sum.push(convertBinaryIntoDecimal(result));
        }
        traversal(root.right, result, sum);
        result.pop();
        return sum;
    }
}

var convertBinaryIntoDecimal = function (result) {
    return result.reduce((acc, cur) => acc * 2 + cur, 0);
}